[Lee Chun Jie] iP

CJ-Lee01 opened the pull request:
# duke.Duke

>You fail to plan you plan to fail - [I don't know who said it](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

- Set Deadlines
- Set ToDos
- Set Events
- Mark and Unmark Tasks.

## What to do?
1. Download it
2. Run it
3. Add your tasks
4. ~~Profit~~ 💹


## Some `code`
```java
class Main {
    public static void main(String[] args) {
        // run Duke.
    }
}
```

## Some checklist
- [ ] bar
- [ ] foo

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

CJ-Lee01 committed:
Added compiled java files to gitignore.

CJ-Lee01 committed:
Changed name of bot.

CJ-Lee01 committed:
Renamed bot back to Duke.

CJ-Lee01 committed:
Added reader to the bot.

CJ-Lee01 committed:
Added StoreList class and enum commands.
For future use.

CJ-Lee01 committed:
Added functionality for level 2.

CJ-Lee01 committed:
Added Task class.

CJ-Lee01 committed:
Done level 3.

CJ-Lee01 committed:
Created new files for tasks.

CJ-Lee01 committed:
Created enum for task creation commands.

CJ-Lee01 committed:
Modified CreateTaskCommands to be more generic.
Implemented subclasses for Task.

CJ-Lee01 committed:
modified Duke to handle more commands.
Updated commands list to add mark and unmark.

CJ-Lee01 committed:
Fixed bug in event creation.

CJ-Lee01 committed:
Added test but idk if they work.

CJ-Lee01 committed:
Implemented error handling.

CJ-Lee01 committed:
Added delete handling.

CJ-Lee01 committed:
Updated test files.

CJ-Lee01 committed:
Shows the enum implementation.

CJ-Lee01 committed:
Add local storage for list.

CJ-Lee01 committed:
Merge branch 'branch-level-7'

CJ-Lee01 committed:
Add datetime support for Deadlines.

CJ-Lee01 committed:
Add datetime functionality for events.

CJ-Lee01 committed:
Add wrapper for parsing datetime.

CJ-Lee01 committed:
Fix datetime bug.

CJ-Lee01 committed:
Add datetime formatting to tasks.

CJ-Lee01 committed:
Clear duke.txt

CJ-Lee01 committed:
Merge branch 'branch-Level-8'

CJ-Lee01 committed:
Add more classes

CJ-Lee01 committed:
Update Duke and StoreList class

CJ-Lee01 committed:
Clean various classes

CJ-Lee01 committed:
Merge branch 'A-MoreOOP'

CJ-Lee01 committed:
Add to duke package

CJ-Lee01 committed:
Merge branch 'A-Packages'

CJ-Lee01 committed:
Move Duke under duke package

CJ-Lee01 committed:
Update files

CJ-Lee01 committed:
Edit files

CJ-Lee01 committed:
Merge branch 'master' into add-gradle-support

CJ-Lee01 committed:
Add tests

CJ-Lee01 committed:
Enforce code style

CJ-Lee01 committed:
Add javadocs

CJ-Lee01 committed:
Add finding function

CJ-Lee01 committed:
Refactor instruction name

CJ-Lee01 committed:
Merge branch 'branch-A-CodingStandard'

CJ-Lee01 committed:
Merge branch 'branch-A-JavaDoc'

CJ-Lee01 committed:
Merge branch 'branch-Level-9'

CJ-Lee01 committed:
Add UI.
followed the tutorial.

CJ-Lee01 committed:
Add GUI.

AprupKale commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+     * @return A string representation of the task in file format.
+     */
+    @Override
+    public String fileString() {
```
Wouldn't getFileString be a better name than fileString for this method?

AprupKale commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,28 @@
+    /** The name of the bot. */
+    private static final String NAME = "Nino!";
+    /**
+     * runs the Duke program.
```
Nitpick: Runs instead of runs

AprupKale commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,102 @@
+        return Task.create(cmd, instructions[1]);
+    }
+
+    public static String findAll(String[] instructions, StoreList src) {
```
Not sure findAll is totally descriptive. findAllTasks seems more descriptive.

AprupKale commented on src/main/java/duke/StoreList.java:
```diff
@@ -0,0 +1,181 @@
+     * @param description The description of the task to be added.
+     * @return A string representing the result of adding the task to the list or an error if unsuccessful.
+     */
+    String add(Commands type, String description) {
```
Missing access specifier!

AprupKale commented on src/main/java/duke/StoreList.java:
```diff
@@ -0,0 +1,181 @@
+     * @param position The position of the task in the list as a string.
+     * @return A string representing the result of marking the task as done.
+     */
+    String markDone(String position) {
```
Missing access specifier!

AprupKale commented on src/main/java/duke/StoreList.java:
```diff
@@ -0,0 +1,181 @@
+     * @param position The position of the task in the list as a string.
+     * @return A string representing the result of marking the task as not done.
+     */
+    String markUndone(String position) {
```
Missing access specifier!

AprupKale commented on src/main/java/duke/StoreList.java:
```diff
@@ -0,0 +1,181 @@
+     * @param position The position of the task in the list as a string.
+     * @return A string representing the result of removing the task from the list.
+     */
+    String delete(String position) {
```
Missing access specifier!

AprupKale commented on src/main/java/duke/UserInterface.java:
```diff
@@ -0,0 +1,149 @@
+        System.out.println(wrapper(s));
+    }
+
+    private static String wrapper(String line) {
```
Can displayWrapper be a better name than wrapper?

AprupKale commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,24 @@
+public class DeadlineTest {
+
+    @Test
+    public void cannotCreateDeadlineWithoutDateTime() {
```
create_withouthDateTime_throwsException follows the convention

AprupKale commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    @Test
+    public void cannotCreateEmptyDeadline() {
```
create_emptyDeadline_throwsException follows the convention

AprupKale commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,18 @@
+public class TodoTest {
+
+    @Test
+    public void cannotCreateEmptyTodo() {
```
create_emptyTodo_throwsException follows the convention

AprupKale commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,24 @@
+public class EventTest {
+
+    @Test
+    public void cannotCreateEventWithoutDateTime() {
```
create_withouthDateTime_throwsException follows the convention

AprupKale commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    @Test
+    public void cannotCreateEmptyEvent() {
```
create_emptyEvent_throwsException follows the convention

CJ-Lee01 replied:
Thank you for the feedback.

AprupKale reviewed (commented):
LGTM! Very good code structure overall. Just minor issues and one nitpick to change.

CJ-Lee01 committed:
Fix MainWindow

CJ-Lee01 committed:
Add Assertions

CJ-Lee01 committed:
Change Task::fileString to Task::toFileString
Changed name of method to improve code quality.

CJ-Lee01 committed:
Fix issue caused by name change
The change from Task::fileString to Task::toFileString broke some code.

CJ-Lee01 committed:
Merge pull request #1 from CJ-Lee01/branch-A-CodeQuality

Branch a code quality

CJ-Lee01 committed:
Merge pull request #2 from CJ-Lee01/branch-A-Assertions

Add Assertions

CJ-Lee01 committed:
Add enum for task attributes.
For future use.

CJ-Lee01 committed:
Add update functionality for tasks.
Command: update <index> <description|deadline|start|end> <updateInfo>
index represents the index shown by the list command.

CJ-Lee01 committed:
Merge branch 'branch-C-Update'

CJ-Lee01 committed:
Merge branch 'master' of https://github.com/CJ-Lee01/ip

CJ-Lee01 committed:
Change UI and add tests
Changed background of UI and photo and added additional tests for the commands and task attributes.

CJ-Lee01 committed:
Fix bugs and add user guide

CJ-Lee01 committed:
Merge branch 'branch-A-UserGuide'

CJ-Lee01 committed:
Add find to documentation

CJ-Lee01 committed:
Fix formatting issues

CJ-Lee01 committed:
Add link in readme

CJ-Lee01 committed:
Fix file access issue

Previous directory used may cause the application to attempt to access
the root folder which caused permission issues.

CJ-Lee01 committed:
Add guide for saving tasks

CJ-Lee01 committed:
Remove redundant part in README
